codec-smtp/src/main/java/io/netty/handler/codec/smtp/DefaultSmtpRequest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.smtp;
+
+import io.netty.util.internal.ObjectUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Default {@link SmtpRequest} implementation.
+ */
+public class DefaultSmtpRequest implements SmtpRequest {
+    private static final int HASH_CODE_PRIME = 31;
+
+    private final SmtpCommand command;
+    private final List<CharSequence> parameters;
+
+    public DefaultSmtpRequest(SmtpCommand command) {
+        this.command = ObjectUtil.checkNotNull(command, "command");
+        parameters = Collections.emptyList();
+    }
+
+    public DefaultSmtpRequest(SmtpCommand command, CharSequence... parameters) {
+        this.command = ObjectUtil.checkNotNull(command, "command");
+        this.parameters = SmtpUtils.toUnmodifiableList(parameters);
+    }
+
+    public DefaultSmtpRequest(CharSequence cmd, CharSequence... parameters) {
+        this(SmtpCommand.valueOf(cmd), parameters);
+    }
+
+    DefaultSmtpRequest(SmtpCommand command, List<CharSequence> parameters) {
+        this.command = ObjectUtil.checkNotNull(command, "command");
+        this.parameters = parameters != null ?
+                Collections.unmodifiableList(parameters) : Collections.<CharSequence>emptyList();
+    }
+
+    @Override
+    public SmtpCommand command() {
+        return command;
+    }
+
+    @Override
+    public List<CharSequence> parameters() {
+        return parameters;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 1;
+        result = HASH_CODE_PRIME * result + command.hashCode();
+        result = HASH_CODE_PRIME * result + parameters.hashCode();
+        return result;
```

Comment:
Could be simplified to `return HASH_CODE_PRIME * command.hashCode() + parameters.hashCode()`